File: api/gen.js
// api/gen.js

const crypto = require('crypto');

// Helper functions
function generateRandomString(length, characters) {
    let result = '';
    for (let i = 0; i < length; i++) {
        result += characters.charAt(Math.floor(Math.random() * characters.length));
    }
    return result;
}

function generateRandomKey(length = 64) {
    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
    return generateRandomString(length, characters);
}

function generateRandomNumber(length = 10) {
    const characters = '0123456789';
    return generateRandomString(length, characters);
}

export default function handler(req, res) {
    const service = req.query.service;

    const baseUrls = {
        "fluxus": "https://flux.li/android/external/start.php?HWID={}",
        "delta": "https://gateway.platoboost.com/a/8?id={}",
        "cryptic": "https://gateway.platoboost.com/a/39097?id={}",
        "evon": "https://pandadevelopment.net/getkey?service=evon&hwid={}",
        "cacti": "https://gateway.platoboost.com/a/23344?id={}",
        "hydrogen": "https://gateway.platoboost.com/a/2569?id={}",
        "relzhub": "https://getkey.relzscript.xyz/redirect.php?hwid={}",
        "vegax": "https://pandadevelopment.net/getkey?service=vegax&hwid={}&provider=linkvertise",
        "deltaios": "https://gateway.platoboost.com/a/2?id={}",
        "arceus": "https://spdmteam.com/key-system-1?hwid={}&zone=Europe/Rome&os=android",
        "trigon": "https://trigonevo.fun/whitelist/?HWID={}-{}-{}-{}-{}"
    };

    if (baseUrls[service]) {
        let modifiedUrl;
        switch (service) {
            case "fluxus":
            case "delta":
            case "cryptic":
            case "deltaios":
            case "cacti":
                modifiedUrl = baseUrls[service].replace('{}', generateRandomKey());
                break;
            case "arceus":
                modifiedUrl = baseUrls[service].replace('{}', generateRandomKey(16));
                break;
            case "evon":
            case "vegax":
                modifiedUrl = baseUrls[service].replace('{}', generateRandomKey(36));
                break;
            case "trigon":
                const k1 = generateRandomKey(6);
                const k2 = generateRandomKey(4);
                modifiedUrl = baseUrls[service].replace('{}', `${k1}-${k2}-${k2}-${k2}-${k1}`);
                break;
            case "hydrogen":
                modifiedUrl = baseUrls[service].replace('{}', generateRandomNumber());
                break;
            case "relzhub":
                modifiedUrl = baseUrls[service].replace('{}', generateRandomKey(20));
                break;
        }
        return res.status(200).json({ url: modifiedUrl });
    }

    return res.status(400).json({ error: "Failed to generate key, service not found." });
}

File: api/gen.py
from urllib.parse import parse_qs

def generate_random_string(length, characters):
    import random
    return ''.join(random.choice(characters) for _ in range(length))

def generate_random_key(length=64):
    characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
    return generate_random_string(length, characters)

def generate_random_number(length=10):
    characters = '0123456789'
    return generate_random_string(length, characters)

def handler(request):
    query = parse_qs(request['queryStringParameters'] or {})
    service = query.get('service', [None])[0]

    base_urls = {
        "fluxus": "https://flux.li/android/external/start.php?HWID={}",
        "delta": "https://gateway.platoboost.com/a/8?id={}",
        "cryptic": "https://gateway.platoboost.com/a/39097?id={}",
        "evon": "https://pandadevelopment.net/getkey?service=evon&hwid={}",
        "cacti": "https://gateway.platoboost.com/a/23344?id={}",
        "hydrogen": "https://gateway.platoboost.com/a/2569?id={}",
        "relzhub": "https://getkey.relzscript.xyz/redirect.php?hwid={}",
        "vegax": "https://pandadevelopment.net/getkey?service=vegax&hwid={}&provider=linkvertise",
        "deltaios": "https://gateway.platoboost.com/a/2?id={}",
        "arceus": "https://spdmteam.com/key-system-1?hwid={}&zone=Europe/Rome&os=android",
        "trigon": "https://trigonevo.fun/whitelist/?HWID={}-{}-{}-{}-{}"
    }

    if service in base_urls:
        if service in ["fluxus", "delta", "cryptic", "deltaios", "cacti"]:
            hwid = generate_random_key()
            url = base_urls[service].format(hwid)
        elif service == "arceus":
            url = base_urls[service].format(generate_random_key(16))
        elif service in ["evon", "vegax"]:
            url = base_urls[service].format(generate_random_key(36))
        elif service == "trigon":
            k1 = generate_random_key(6)
            k2 = generate_random_key(4)
            url = base_urls[service].format(k1, k2, k2, k2, k1)
        elif service == "hydrogen":
            url = base_urls[service].format(generate_random_number())
        elif service == "relzhub":
            url = base_urls[service].format(generate_random_key(20))
        return {
            "statusCode": 200,
            "body": f'{{"url": "{url}"}}',
            "headers": { "Content-Type": "application/json" }
        }

    return {
        "statusCode": 400,
        "body": '{"error": "Invalid service"}',
        "headers": { "Content-Type": "application/json" }
    }
